hlf/chaincode/admission.md

```diff
@@ -0,0 +1,227 @@
+# Hyperledger Admission Api
+
+In the following, the chaincode api for the "Admission"-Chaincode is described.
+This contains all Transactions and Models provided for this domain.
+The Errors returned are defined [here](errors.md#Errors).
+
+## Transactions
+
+### Add Admission
+- ID = addAdmission
+- Send
+    - enrollmentId :: String
+    - courseId :: String
+    - moduleId :: String
+    - timestamp :: String \<DATE YYYY-MM-DD_hh:mm\>

```

Comment:
This is a strange date format

Reply:
> We should use a more standard format for timestamps, e.g. ISO 8601, the unix time stamp or similar.
> 
> I am not sure about the new HLInvalid Error. Why do you prefer to use that over the usual validation error with invalid params? To me, they seem to serve the same purpose, except that instead of directly showing me that the moduleId is the problem, I would need to create a mapping from "ModuleAccess" to "moduleId" to know where I need to show the validation error.

1. ISO 8601 - noted, will change according to that, thank you
2. Problem with mapping this to the existing format is: we attribute errors to certain parameters. The error in question does not directly map to just the moduleId, but to the combination of moduleId & enrollmentId. I am open to suggestions though. Now that I think about it with a refreshed mind, when submitting an admission, we usually can assume that the enrollmentId will not be changed when the sender fixed their error. The most "common" mistake will be attributing the course to a wrong module. Thus in this case we might be able to map it to the InvalidParams Error, too. A discussion might help here. Will there be future cases, where the error is not directly related to a single parameter?

Reply:
solved in https://github.com/upb-uc4/api/pull/83/commits/b99ee2ba31c3d2b7443cd6c99e99e474c01a0985